### DISCUSSION TITLE
Get visible row data in viewport

### DISCUSSION THREAD

**Discussion by blessenabrahamnw18** [komarovalexander/ka-table#323] [General]
Hi @komarovalexander 
I have table with 500 rows, only 10 - 12 are visible at a time.
Is there a way to get row data for visibile rows in viewport, once user scrolls and the 
viewport is updated with visible rows need updated entries of rowdata.
Is there any event like "areAllVisibleRowsSelected" to get the "visiblerows"

Thanks

**Comment by komarovalexander** [comment]
Hi check this https://ka-table.com/docs_props.html#kapropsutils

**Reply by blessenabrahamnw18** [reply]
yes, I did check kaproputils; "areAllVisibleRowsSelected" was the closest event available to what I wanted, but it is for selected data and returns only true or false;
I don't have pagination and no selection event involved; I need to call API for visible rows in the viewport out of all rows in ka-table and update the result using Updatecellvaule.

**Reply by komarovalexander** [reply]
Check this https://github.com/komarovalexander/ka-table/blob/master/src/lib/Components/VirtualizedRows/VirtualizedRows.tsx I think you can get virtualizedData the same way
